### PR TITLE
support for python 3.9

### PR DESCRIPTION
**Pull request by krober10nd**
* removes installation blocker for python3.9
* localizes segyio dependency installation into the only local function that uses it.

### CODE REVIEW

**Comment by nschloe** [comment]
Make the dependency optional, too. See https://github.com/nschloe/meshio/blob/main/setup.cfg#L44.

**Comment by nschloe** [comment]
You may want to lose `python_requires=">=3.0"` from setup.py too. (It's already in setup.cfg.)

File: setup.py [hunk 1]
```diff
@@ -95,5 +97,6 @@ def build_extension(self, ext):
     cmdclass=dict(build_ext=CMakeBuild),
     extras_require={
         "benchmarking": benchmarking,
+        "all": segyio,
```
Comment (nschloe) [inline setup.py:100]: If you put it under "all", I'd add benchmarking to the list, too. In any case, I'd always move this to setup.cfg.

**Comment by nschloe** [comment]
> so I put it into the setup.py file but same issue.

setup.py overrides setup.cfg. Try putting it all into setup.cfg.

**Comment by krober10nd** [comment]
alright, I created three optionals: `io`, `benchmarking`, and `all` which is a combination of `io` and `benchmarking`. I didn't want to the other packages that I use for a mesh generation comparison coming into to common installs as these require other additional third party packages (like gmsh).  I put all these optionals in setup.cfg and tested this on a virtual env with python 3.9. I also added another CI build for 3.9.

I plan to improve io capabilities here in some time.

**Pull request merged**